packages/opds-web-client/src/OPDSDataAdapter.ts

```diff
@@ -118,12 +119,26 @@ export function entryToBook(entry: OPDSEntry, feedUrl: string): BookData {
     borrowUrl = resolve(feedUrl, borrowLink.href);
   }
 
-  let allBorrowLinks = <OPDSAcquisitionLink>entry.links.filter(link => {
-    return (
-      link instanceof OPDSAcquisitionLink &&
-      link.rel === OPDSAcquisitionLink.BORROW_REL
-    );
-  });
+  let allBorrowLinks: FulfillmentLink[] = entry.links
+    .filter(link => {
+      return (
+        link instanceof OPDSAcquisitionLink &&
+        link.rel === OPDSAcquisitionLink.BORROW_REL
+      );
+    })
+    .map(link => {
```

Comment:
This part is new. What's this doing?

Reply:
Yeah 
https://github.com/NYPL-Simplified/opds-web-client/pull/277/files/7588148a1a90dea7f8b80cb9d35b24d9fb29b5e6..ed35a5b0224aca1313cf9d8d2cdbf47ed9ace736#diff-60c3f73af73e89b9faef5deab53edec4R63

So this wasn't actually typechecking before and the thing that was being returned was an object of type `OPDSAcquisitionLink` so what this is doing is returning a type of `FulfillmentLink` -- if you look at the FulfillmentLinks stuff before, this code is pretty much the same.  

I didn't extract it because I don't think that there's a reason these implementations should be tied together.  

But the idea is that all the info I need is in the `FulfillmentLink` type.  `OPDSAcquisitionLink` is a lot of extraneous information that just makes the `BookData` object harder to mock in CPW.